src/Sylius/Component/Product/Model/Product.php

```diff
@@ -247,6 +248,12 @@ public function getAttributes()
      */
     public function addAttribute(AttributeValueInterface $attribute)
     {
+        Assert::isInstanceOf(
+            $attribute,
+            'Sylius\Component\Product\Model\ProductAttributeValueInterface',
```

Comment:
I thought of it, but I was thinking it was PHP7 only ; I was wrong, I'll fix that in a minute ;)